components/style/properties/gecko.mako.rs

```diff
@@ -1338,17 +1361,7 @@ ${impl_style_struct(style_struct)}
 ${define_ffi_struct_accessor(style_struct)}
 % endfor
 
-lazy_static! {
-    pub static ref INITIAL_GECKO_VALUES: ComputedValues = ComputedValues {
-        % for style_struct in data.style_structs:
-           ${style_struct.ident}: style_structs::${style_struct.name}::initial(),
-        % endfor
-        custom_properties: None,
-        shareable: true,
-        writing_mode: WritingMode::empty(),
-        root_font_size: longhands::font_size::get_initial_value(),
-    };
-}
+static mut INITIAL_GECKO_VALUES: *mut ComputedValues = 0 as *mut ComputedValues;
 
```

Comment:
How could we create it (without a `lazy_static!`) statically without
const fns?

On 08/04/2016 04:32 PM, Eduard-Mihai Burtescu wrote:
>> @@ -1338,17 +1361,7 @@ ${impl_style_struct(style_struct)}
>>  ${define_ffi_struct_accessor(style_struct)}
>>  % endfor
>>  
>> -lazy_static! {
>> -    pub static ref INITIAL_GECKO_VALUES: ComputedValues = ComputedValues {
>> -        % for style_struct in data.style_structs:
>> -           ${style_struct.ident}: style_structs::${style_struct.name}::initial(),
>> -        % endfor
>> -        custom_properties: None,
>> -        shareable: true,
>> -        writing_mode: WritingMode::empty(),
>> -        root_font_size: longhands::font_size::get_initial_value(),
>> -    };
>> -}
>> +static mut INITIAL_GECKO_VALUES: *mut ComputedValues = 0 as *mut ComputedValues;
>>  
> 
> You can at least use an `AtomicUsize` and wrap that in a safe interface.
> 
> ---
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub:
> https://github.com/servo/servo/pull/12725/files/b916ad0a85b88eb40d919986cdc0c6b7420a4fab#r73621079
>